libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/Agent.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
+
+import java.lang.instrument.Instrumentation;
+import java.nio.channels.SocketChannel;
+import java.util.Map;
+
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.ClassFileLocator;
+import net.bytebuddy.dynamic.loading.ClassInjector;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.matcher.ElementMatcher.Junction;
+import net.bytebuddy.matcher.ElementMatchers;
+
+/**
+ * Java Agent
+ */
+public class Agent {
+    /**
+     * Constructor
+     */
+    private Agent() {}
+
+    /**
+     * Premain
+     * @param agentArguments agent arguments
+     * @param instrumentation instrumentation
+     * @throws Exception Exception
+     */
+    public static void premain(String agentArguments, Instrumentation instrumentation) throws Exception {
+        initAgent(instrumentation);
+    }
+
+    /**
+     * Agent Main
+     * @param agentArguments agent arguments
+     * @param instrumentation instrumentation
+     * @throws Exception Exception
+     */
+    public static void agentmain(String agentArguments, Instrumentation instrumentation) throws Exception {
+        initAgent(instrumentation);
+    }
+
+    private static AgentBuilder createAgentBuilder(Instrumentation inst) throws Exception {
+        final Junction<TypeDescription> systemType = ElementMatchers.isSubTypeOf(SocketChannel.class);
```

Comment:
If I understand correctly, this will intercept all calls to all methods of SocketChannel from any thread ( `.ignore(ElementMatchers.none())` ) in the OS JVM. 

So, does it mean that we have to specify allow/deny list for all other things or probably do this at  `java.nio.channels.Channel` and `java.net.Socket` level ? 

Essentially, we have to maintain the mapping with all parent classes that can be used for N/W Connections -> All/subset of things covered under SocketPermission by JSM and keep it updated with every Java upgrade.

Reply:
> So, does it mean that we have to specify allow/deny list for all other things or probably do this at `java.nio.channels.Channel` and `java.net.Socket` level ?

Correct @expani , basically this is just a POC that focuses on one single flow (`SocketChannel::connect`), but yes, in general - we would need to include more (and also cover file system access, at the minimum)

Reply:
@kumargu Other things are the constants that @reta  has defined in `SecurityConstants` 

We would have to identify the Java core classes that might come in the path for utilizing that permission. 

For instance, SocketPermission will be applicable to core classes `java.nio` and `java.net` similarly for FilePermission, ClassLoaderPermission; etc. 

We might not be targeting it for OS 3.0 but it would eventually grow into it.

Reply:
> @kumargu Other things are the constants that @reta has defined in `SecurityConstants`

@expani so this (and other classes in the `org.opensearch.secure_sm.policy` package) are copied from JDK (pre JDK-24) in order to have an ability to parse Java policies (it is also gone in JDK-24). We are aiming for very limited instrumentation, even in long term.

Reply:
@expani I would also like to make clear that we are not doing any class-loader-protection for the deprication in 3.0. So keeping that aside for our discussion will help in clear conversations. 

refer table for more details: https://github.com/opensearch-project/OpenSearch/issues/17181#issuecomment-2625330578